src/kaldi_recognizer.cc

```diff
@@ -397,7 +397,8 @@ bool KaldiRecognizer::GetSpkVector(Vector<BaseFloat> &out_xvector, int *num_spk_
     //  xvector_result is filled with xvector for PldaScoring process
     xvector_result = xvector;
     //  out_xvector will be filled by PldaScoring method from utterance
-    //      xvector after transformation
+    //      xvector before transformation so that it can be used for new
+    //      users enrollment
     PldaScoring(out_xvector);
```

Comment:
Why you get this out_xvector? only for enrollment? I guess that maybe it's better to add a specific method for this task.

Reply:
`out_xvector` is passed as reference to this function and it will be filled to sending to user as `spk` field.
Yes it can be used for enrollment and it should be added to `spk_xvectors.ark` as ark format.
Defining new function will make some redundancy computing xvector again while we compute this values in `PldaScoring` once.

Reply:
It may be best to review the entire speaker enrollment and scoring scenario once. Using a function for two purposes is not interesting.